src/main/java/com/google/ehub/data/EntertainmentItemDatastore.java

```diff
@@ -0,0 +1,104 @@
+  public static final String TITLE_PROPERTY_KEY = "title";
+  public static final String LOWERCASE_TITLE_PROPERTY_KEY = "lowercaseTitle";
+  public static final String DESCRIPTION_PROPERTY_KEY = "description";
+  public static final String IMAGE_URL_PROPERTY_KEY = "imageURL";
+
+  private static DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+
+  /**
+   * Adds an EntertainmentItem to Datastore.
+   *
+   * @param title title of the entertainment item
+   * @param description description of the entertainment item
+   * @param imageURL URL to the image of the entertainment item stored in Blobstore
+   */
+  public static void addEntertainmentItemToDatastore(
+      String title, String description, String imageURL) {
+    Entity entertainmentItemEntity = new Entity(ENTERTAINMENT_ITEM_KIND);
+
+    entertainmentItemEntity.setProperty(TITLE_PROPERTY_KEY, title);
+    entertainmentItemEntity.setProperty(LOWERCASE_TITLE_PROPERTY_KEY, title.toLowerCase());
+    entertainmentItemEntity.setProperty(DESCRIPTION_PROPERTY_KEY, description);
+    entertainmentItemEntity.setProperty(IMAGE_URL_PROPERTY_KEY, imageURL);
+
+    datastoreService.put(entertainmentItemEntity);
+  }
+
+  /**
+   * Queries all entertainment items found in Datastore.
+   *
+   * @return list with all entertainment items found in Datastore
+   */
+  public static List<EntertainmentItem> queryAllEntertainmentItems() {
+    return getListFromQuery(new Query(ENTERTAINMENT_ITEM_KIND));
+  }
+
+  /**
+   * Queries all entertainment items found in Datastore and orders them based on title with the
+   * given sorting direction
+   */
+  public static List<EntertainmentItem> queryAllEntertainmentItemsWithTitleOrder(
+      SortDirection sortDirection) {
+    return getListFromQuery(
+        new Query(ENTERTAINMENT_ITEM_KIND).addSort(LOWERCASE_TITLE_PROPERTY_KEY, sortDirection));
+  }
+
+  /**
+   * Queries entertainment items with the specified title prefix and sorting direction.
+   *
+   * @param title the title prefix used to filter the query
+   * @param sortDirection the sort direction used to order the entertainment items based on title
+   * @return list with the entertainment items that match the title prefix and sorting direction
+   */
+  public static List<EntertainmentItem> queryEntertainmentItemsByTitle(
+      String title, SortDirection sortDirection) {
+    return getListFromQuery(
+        new Query(ENTERTAINMENT_ITEM_KIND)
+            .addSort(LOWERCASE_TITLE_PROPERTY_KEY, sortDirection)
+            .setFilter(CompositeFilterOperator.and(
+                new FilterPredicate(LOWERCASE_TITLE_PROPERTY_KEY,
+                    FilterOperator.GREATER_THAN_OR_EQUAL, title.toLowerCase()),
+                new FilterPredicate(LOWERCASE_TITLE_PROPERTY_KEY, FilterOperator.LESS_THAN,
+                    NextIncreasingLexicographicStringUtility.getNextIncreasingLexicographicString(
+                        title.toLowerCase())))));
```

Comment:
Can you use FilterOperator.LESS_THAN_OR_EQUAL with title.toLowerCase() instead?

Reply:
For the prefix search to work I still have to find the next lexicographic string so my upper bound for searches is not the current title but the next string that does not share the same prefix. I tried to search if there is a simpler way to do it in Datastore but haven't found any other solutions.